### DISCUSSION TITLE
Use NuShell as a library

### DISCUSSION THREAD

**Discussion by J-Cake** [nushell/nushell#9728] [Q&A]
Hiya! Love NuShell generally, but I'd like to be able to invoke NuShell from within a program I'm writing which will focus heavily on customisable actions, for which I'd like to employ NuShell. The difference is that I'd like to be able to a) invoke NuShell without spawning a separate process and b) send/retrieve pre-structured data. For example, if the user enters a script which returns a table, I'd like to able to work with that table, without having to parse it.

Any suggestions?

**Comment by fdncred** [comment]
You might check out this https://github.com/nushell/awesome-nu#integrations. couchbase-shell is integrated nushell tightly but it's not quite what you're describing.

**Reply by J-Cake** [reply]
Hm not quite. Thank you though, it's a good starting point, but I'll keep looking
